gateway/src/service.rs

```diff
@@ -835,4 +866,51 @@ pub mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn service_create_destroy_recreate_project() -> anyhow::Result<()> {
+        let world = World::new().await;
+        let svc = Arc::new(GatewayService::init(world.args(), world.pool()).await);
+
+        let account: AccountName = "neo".parse().unwrap();
+        let project_name: ProjectName = "matrix".parse().unwrap();
+        let domain: FQDN = "neo.the.matrix".parse().unwrap();
+        let certificate = "dummy certificate";
+        let private_key = "dummy private key";
+
+        svc.create_user(account.clone()).await.unwrap();
+
+        assert_err_kind!(
+            svc.project_details_for_custom_domain(&domain).await,
+            ErrorKind::CustomDomainNotFound
+        );
+
+        let _ = svc
+            .create_project(project_name.clone(), account.clone())
+            .await
+            .unwrap();
+
+        svc.create_custom_domain(project_name.clone(), &domain, certificate, private_key)
+            .await
+            .unwrap();
+
+        let mut work = svc
+            .new_task()
+            .project(project_name.clone())
+            .and_then(task::destroy())
+            .build();
+
+        while let TaskResult::Pending(_) = work.poll(()).await {}
+        assert!(matches!(work.poll(()).await, TaskResult::Done(())));
+
+        let recreated_project = svc
+            .create_project(project_name.clone(), account.clone())
+            .await
+            .unwrap();
+
+        assert!(matches!(recreated_project, Project::Creating(_)));
+        // TODO: check `recreated_project->fqdn == domain`
```

Comment:
Possibly related to the issue above on how to instantiate a project. There is no way to check that the `fqdn` field in ProjectCreating is as expected since it's private. Could add a `#[cfg(test)]` getter but I haven't seen it done elsewhere in the project,

Reply:
Making the field `pub(crate)` will also be acceptable

Reply:
I ended up making it a `cfg(test)` getter, I reckon it's more suggestive of its expected usage (and also ends up behaving as `pub(crate)` anyway).

Reply:
That makes sense, it shouldn't be more public than it needs to be. We can always make it pub(crate) in the future if we need it outside of tests. :ok_hand: